sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SymbolicExecution/ExplodedGraphTests.cs

```diff
@@ -935,5 +935,93 @@ public void Test(string s)
 
             explodedGraph.Walk();
         }
+
+        [TestMethod]
+        [TestCategory("Symbolic execution")]
+        public void ExplodedGraph_DefaultLiteral()
+        {
+            var testInput = "var i = default(int); int j = default; System.IO.File k = default;";
+            var method = ControlFlowGraphTest.CompileWithMethodBody(string.Format(TestInput, testInput), "Bar", out var semanticModel);
+            var methodSymbol = semanticModel.GetDeclaredSymbol(method);
+            var varDeclarators = method.DescendantNodes().OfType<VariableDeclaratorSyntax>();
+            var iSymbol = semanticModel.GetDeclaredSymbol(varDeclarators.First(d => d.Identifier.ToString() == "i"));
+            var jSymbol = semanticModel.GetDeclaredSymbol(varDeclarators.First(d => d.Identifier.ToString() == "j"));
+            var kSymbol = semanticModel.GetDeclaredSymbol(varDeclarators.First(d => d.Identifier.ToString() == "k"));
+
+            var cfg = CSharpControlFlowGraph.Create(method.Body, semanticModel);
+            var lva = CSharpLiveVariableAnalysis.Analyze(cfg, methodSymbol, semanticModel);
+            var explodedGraph = new CSharpExplodedGraph(cfg, methodSymbol, semanticModel, lva);
+
+            var numberOfProcessedInstructions = 0;
+            explodedGraph.InstructionProcessed +=
+                (sender, args) =>
+                {
+                    numberOfProcessedInstructions++;
+                    if (args.Instruction.ToString() == "i = default(int)")
```

Comment:
- we transform the instruction to string multiple times, an option would be to do it once and use the value in the if cases.
- we could also use a switch since (I find the switch more readable)

Reply:
Added a variable to hold the string value of the instruction. I'd rather no use switch here when there is only a few (3) cases.
Same as for the other test, as I find the comments are necessarys, and adding them to a switch expression makes it a bit less readable in my opinion. (But this one I don't mind updating if you think it better as a switch)